File: disturbances/src/main/java/md/vnastasi/cloud/endpoint/model/Disruption.java
package md.vnastasi.cloud.endpoint.model;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.List;

public record Disruption(
        @JsonProperty("id")
        String id,

        @JsonProperty("title")
        String title,

        @JsonProperty("type")
        DisturbanceType type,

        @JsonProperty("period")
        Period period,

        @JsonProperty("intervals")
        List<Interval> intervals,

        @JsonProperty("isActive")
        boolean isActive
) {
}

File: disruptions/src/main/kotlin/md/vnastasi/cloud/endpoint/model/Disruption.kt
package md.vnastasi.cloud.endpoint.model

import com.fasterxml.jackson.annotation.JsonFormat
import java.time.OffsetDateTime

data class Disruption(
    val id: String,
    val summary: String,
    val cause: String,
    val expectation: String,
    val alternativeTransport: String,
    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss.SSSX")
    val reportedAt: OffsetDateTime
)
